providers/oidc.go

```diff
@@ -120,11 +120,12 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 		return nil, fmt.Errorf("email in id_token (%s) isn't verified", claims.Email)
 	}
 
+	// Use idToken.Expiry because that's what we verify against
 	return &SessionState{
 		AccessToken:  token.AccessToken,
 		IDToken:      rawIDToken,
 		RefreshToken: token.RefreshToken,
-		ExpiresOn:    token.Expiry,
+		ExpiresOn:    idToken.Expiry,
```

Comment:
We want to use this expiry here because it guarantees that a token is refreshed on expiration, so we will always be sure to pass a valid ID token along when `-pass-authorization-header` is set.

Reply:
I _think_ this is the right thing to do, was there not an issue related to this?

Reply:
I didn't find an issue filed for this, but I think it's correct as well.